pkg/ibmvpc/parser.go

```diff
@@ -1027,12 +789,18 @@ func getSubnetByCidr(m map[string]*Subnet, cidr string) (*Subnet, error) {
 
 func getIKSnodesConfig(res map[string]*vpcmodel.VPCConfig,
 	subnetNameToSubnet map[string]*Subnet,
-	rc *ResourcesContainer,
-	skipByVPC func(string) bool) {
-	for _, iksNode := range rc.iksNodes {
-		subnet, err := getSubnetByCidr(subnetNameToSubnet, iksNode.Cidr)
+	rc *datamodel.ResourcesContainerModel,
+	skipByVPC func(string) bool) error {
+	for _, iksNode := range rc.IKSWorkerNodes {
+		if len(iksNode.NetworkInterfaces) != 1 {
+			return errIksParsing
+		}
+		iksNodeNetIntfObj := iksNode.NetworkInterfaces[0]
+
+		subnet, err := getSubnetByCidr(subnetNameToSubnet, *iksNodeNetIntfObj.Cidr)
```

Comment:
Question: should we be concerned that subnet name is not unique across VPCs? Should we better use CRN?

Reply:
right, opened issue #324